src/invite/hooks.tsx

```diff
@@ -0,0 +1,35 @@
+import { isNil } from 'lodash'
+import { useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { Share } from 'react-native'
+import { useSelector } from 'react-redux'
+import { InviteEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { FIREBASE_ENABLED } from 'src/config'
+
+import { walletAddressSelector } from 'src/web3/selectors'
+
+import { createDynamicLink } from './utils'
+
+export function useShareUrl() {
+  const [shareUrl, setShareUrl] = useState<string | null>(null)
+
+  const address = useSelector(walletAddressSelector)
+  const { t } = useTranslation()
+
+  const message = isNil(shareUrl) ? null : t('inviteWithUrl.share', { shareUrl })
+
+  useAsync(async () => {
+    if (isNil(address) || !FIREBASE_ENABLED) return
+    const url = await createDynamicLink(address)
```

Comment:
we should probably handle the case where this fails, even if it is a very edge case. i'd propose to log the error using Logger.debug so that we can debug if users contact support. perhaps showing the global error banner is also a good idea (you can do something like `dispatch(showError(...))`

Reply:
Sounds good to me, I'll add that today.

Reply:
Done. I'm open to feedback on copy and Logger error formatting.